ng add @angular/ssr not working anymore after upgrade to v20

### Which @angular/* package(s) are the source of the bug?

upgrade, Don't known / other

### Is this a regression?

Yes

### Description


I upgraded this project to v20 to demo SSR for a public talk: https://github.com/alcfeoh/ngx-training/tree/ssr-workshop

And after running `ng add @angular/ssr`, the project doesn't compile anymore, and `ng serve` throws an error.


### Please provide a link to a minimal reproduction of the bug

https://github.com/alcfeoh/ngx-training/tree/ssr-upgrade-error

### Please provide the exception or error you saw

```true
ng serve
Application bundle generation failed. [1.638 seconds]

✘ [ERROR] File 'src/main.server.ts' is missing from the TypeScript compilation. [plugin angular-compiler]

    angular:main-server:angular:main-server:8:24:
      8 │ export { default } from './src/main.server';
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 20.0.2
Node: 22.16.0
Package Manager: npm 10.9.2
OS: darwin arm64

Angular: 20.0.3
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.2000.2
@angular-devkit/core         18.2.7
@angular-devkit/schematics   20.0.2
@angular/build               20.0.2
@angular/cli                 20.0.2
@angular/ssr                 20.0.2
@schematics/angular          20.0.2
rxjs                         7.8.2
typescript                   5.8.3
zone.js                      0.15.1
```

### Anything else?

_No response_

--- a/packages/schematics/angular/server/index.ts
+++ b/packages/schematics/angular/server/index.ts
@@ -119,6 +119,14 @@ function updateConfigFileApplicationBuilder(options: ServerOptions): Rule {
 function updateTsConfigFile(tsConfigPath: string): Rule {
   return (host: Tree) => {
     const json = new JSONFile(host, tsConfigPath);
+    // Skip adding the files entry if the server entry would already be included.
+    const include = json.get(['include']);
+    if (!Array.isArray(include) || !include.includes('src/**/*.ts')) {
+      const filesPath = ['files'];
+      const files = new Set((json.get(filesPath) as string[] | undefined) ?? []);
+      files.add('src/' + serverMainEntryName);
+      json.modify(filesPath, [...files]);
+    }
 
     const typePath = ['compilerOptions', 'types'];
     const types = new Set((json.get(typePath) as string[] | undefined) ?? []);

